### PR TITLE
Book Index Authors

### LINKED ISSUE #3: User Story 6, Book Index Page
```
As a Visitor,
When I visit a book index page,
I see all book titles in the database.
Each book entry on the page shows:
- the author(s) who wrote the book
- the number of pages in the book
- and the year it was published
- a small thumbnail image of the book cover
```

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #26

child of #33

### PR DESCRIPTION
**Pull request by alexander-mathieu**
- Add authors to book index page with spec
- Add author_names instance method to Book model with spec
- Closes #3

### CODE REVIEW

File: app/views/books/index.html.erb [hunk 1]
```diff
@@ -4,6 +4,7 @@
   <section class="book" id="book-<%= book.id %>">
     <img src="<%= book.image %>" alt="<%= book.title %> Cover">
     <h5><%= book.title %></h5>
+    <p><%= book.author_names.join(", ") %></p>
```
Comment (BrennanAyers) [inline app/views/books/index.html.erb:7]: Presentation logic!

File: spec/features/books/index_spec.rb [hunk 2]
```diff
@@ -3,9 +3,15 @@
 RSpec.describe 'As a user', type: :feature do
   describe 'When I visit /books' do
     before :each do
-      @book_1 = Book.create!(title: "Veronica Mars", pages: 10, year: 2012)
-      @book_2 = Book.create!(title: "Mars Aquatic", pages: 120, year: 1964)
-      @book_3 = Book.create!(title: "Trip to Mars", pages: 480, year: 2020)
+      @author_1 = Author.create!(name: "Brennan Ayers")
+      @author_2 = Author.create!(name: "John Flapjacks")
+      @author_3 = Author.create!(name: "Patrick Duvall, M.D.")
```
Comment (BrennanAyers) [inline spec/features/books/index_spec.rb:8]: Moving up!

File: app/models/book.rb [hunk 3]
```diff
@@ -4,4 +4,8 @@ class Book < ApplicationRecord
   has_many :authors, through: :books_by_author
 
   validates_presence_of :title, :pages, :year
+
+  def author_names
+    authors.pluck(:name)
```
Comment (BrennanAyers) [inline app/models/book.rb:9]: 👍
